compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1787,8 +1819,39 @@ impl<'a> Parser<'a> {
                 Ok(true) => {}
                 Ok(false) => unreachable!(),
                 Err(mut err) => {
+                    // Qualifier keywords ordering check
+                    // Extern is not checked since it is the last expected one
+                    let misplaced_qualifier =
+                        [kw::Const, kw::Async, kw::Unsafe].iter().any(|&k| self.check_keyword(k));
+
+                    if misplaced_qualifier {
```

Comment:
This test seems superfluous – the if-chain below to compute `current_qual_sp` does the same thing immediately after. Removing this check will also allow you to get rid of the unnecessary `unreachable!()`.